documentation/homework/week1/src/components/todoAdd.js

```diff
@@ -0,0 +1,40 @@
+import React,{Component} from 'react';
+import DatePicker from 'react-datepicker';
+import moment from 'moment';
+import 'react-datepicker/dist/react-datepicker.css';
+
+class TodoAdd extends Component{
+    state = {
+        id:'',
+        description:'',
+        deadline:moment(),
+        done:false
+    }
+    onChangeState(field,value){
+        const currentState=this.state;
```

Comment:
Taking state in a variable like this, means you takes the reference of your state on `currentState` and changing anything using this reference will end up changing your original `state` itself (and we should NOT do this at all).

Instead you can rewrite this `onChangeState` like the follows

`this.setState((state) =>({...state, [field]: value}));`
or
`this.setState({...this.state, [field]: value})`

This expression `{...this.state}` takes all the properties on your state object and write it again in a totally new object.

If you don't know this syntax for es6 `{...this.state}` which ((destructuring assignment)) you can use `Object.assign({}, this.state, {[field]: value})`

**Note**: you also can use `this.setState({[field]: value})` without returning the full state.